File: Python/Bits/entierEnBinaire.py
def entierEnBinaire(n: int) -> str:
    """
    Description:
        Convertit un entier en binaire. Pour gérer les nombres négatifs, on ajoute 2**32 (ou 1<<32).

    Paramètres:
        n: {int} -- L'entier à convertir.

    Retourne:
        {str} -- Lé nombre converti en binaire.

    Exemple:
        >>> entierEnBinaire(5)
        00000000000000000000000000000101

        >>> entierEnBinaire(-5)
        11111111111111111111111111110111    
    """
    return bin(n + (1<<32))[3:]

File: Haskell/Bits/entierEnBinaire.hs
module EntierEnBinaire (entierEnBinaire) where

import Data.Bits (shiftR, (.&.))
{-| Fonction qui convertit un entier en binaire.
    (x .&. 1) pour récupérer le bit de poids faible.
    On ajoute 10 à x pour que le résultat soit toujours un entier, 
    et on applique la fonction récursive sur le reste de x, en divisant par 2.

    Input:  Int

    Output: Int
-}
entierEnBinaire :: Int -> Int
entierEnBinaire 0 = 0
entierEnBinaire x = (x .&. 1) + 10 * entierEnBinaire (shiftR x 1)

-- TEST
main :: IO ()
main = do
    mapM_ 
        (\x -> putStrLn $ "entierEnBinaire " ++ show x ++ " -> " ++ show (entierEnBinaire x))
        [0..10]
{-^ Output:
    entierEnBinaire 0 -> 0
    entierEnBinaire 1 -> 1
    entierEnBinaire 2 -> 10
    entierEnBinaire 3 -> 11
    entierEnBinaire 4 -> 100
    entierEnBinaire 5 -> 101
    entierEnBinaire 6 -> 110
    entierEnBinaire 7 -> 111
    entierEnBinaire 8 -> 1000
    entierEnBinaire 9 -> 1001
    entierEnBinaire 10 -> 1010
-}